overlord/servicestate/service_control.go

```diff
@@ -149,6 +149,11 @@ func (m *ServiceManager) doServiceControl(t *state.Task, _ *tomb.Tomb) error {
 			if err != nil {
 				return err
 			}
+
+			// QueryDisabledServices returns snap's service (app) names, we need to
+			// convert them to systemd unit names for proper matching below.
+			snapDisabledServices = wrappers.GenServiceNames(info, snapDisabledServices)
```

Comment:
This is incorrect because wrappers.StartServices() below expects app names for disabled svcs.